plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,270 @@
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.consul import (
+    AUTH_ARGUMENTS_SPEC,
+    OPERATION_CREATE,
+    OPERATION_UPDATE,
+    OPERATION_DELETE,
+    OPERATION_READ,
+    _ConsulModule,
+    camel_case_key,
+    RequestError,
+    validate_check,
+)
+
+_ARGUMENT_SPEC = {
+    "state": dict(default="present", choices=["present", "absent"]),
+    "name": dict(type='str'),
+    "id": dict(type='str'),
+    "interval": dict(type='str'),
+    "notes": dict(type='str'),
+    "http": dict(type='str'),
+    "tcp": dict(type='str'),
+    "ttl": dict(type='str'),
+    "timeout": dict(type='str'),
+    "service_id": dict(type='str'),
+}
+
+_MUTUALLY_EXCLUSIVE = [
+    ('args', 'ttl', 'tcp', 'http'),
+]
+
+_REQUIRED_IF = [
+    ('state', 'present', ['name']),
+    ('state', 'absent', ['id']),
+]
+
+_REQUIRED_BY = {
+    'args': 'interval',
+    'http': 'interval',
+    'tcp': 'interval',
+}
+
+_ARGUMENT_SPEC.update(AUTH_ARGUMENTS_SPEC)
+
+
+class ConsulAgentCheckModule(_ConsulModule):
+    api_endpoint = "agent/check"
+    result_key = "check"
+    unique_identifier = "id"
+
+    def endpoint_url(self, operation, identifier=None):
+        if operation == OPERATION_READ:
+            return "/".join([self.api_endpoint + 's'])
+        if operation in [OPERATION_CREATE, OPERATION_UPDATE]:
+            return "/".join([self.api_endpoint, "register"])
+        if operation == OPERATION_DELETE:
+            return "/".join([self.api_endpoint, "deregister", identifier])
+
+        return super(ConsulAgentCheckModule, self).endpoint_url(operation, identifier)
+
+    def read_object(self):
+        url = self.endpoint_url(OPERATION_READ)
+        checks = self.get(url)
+
+        if self.params.get(self.unique_identifier) in checks:
+            return checks[self.params.get(self.unique_identifier)]
+        else:
+            return
+
+    def prepare_object(self, existing, obj):
+        existing = super(ConsulAgentCheckModule, self).prepare_object(existing, obj)
+
+        operational_attributes = {"Node", "CheckID", "Output", "ServiceName", "ServiceTags",
+                                  "Status", "Type", "ExposedPort", "Definition"}
+        existing = {
+            k: v for k, v in existing.items() if k not in operational_attributes
+        }
+
+        validate_check(existing)
+
+        if 'Args' in existing and existing['Args']:
+            existing['Args'] = ["sh", "-c", existing['Args']]
+
+        return existing
+
+    def create_object(self, obj):
+        super(ConsulAgentCheckModule, self).create_object(obj)
+        if self._module.check_mode:
+            return obj
+        else:
+            return self.read_object()
+
+    def update_object(self, existing, obj):
+        super(ConsulAgentCheckModule, self).update_object(existing, obj)
+        if self._module.check_mode:
+            return obj
+        else:
+            return self.read_object()
```

Comment:
Same as above.

Reply:
Same as above, but when we want to update a Service/Check.